src/views/orderReport.tsx

```diff
@@ -0,0 +1,467 @@
+import { CURRENT_PST_DATE } from "../utils/dateTimeUtil";
+import { fetchData } from "../utils/fetchUtil";
+import { getTableHeaders } from "../components/utils/Utils";
+
+const OrderReport = () => {
+  const [isLoading, setIsLoading] = useState<boolean>(false);
+  const [hourlyReport, setHourlyReport] = useState<OrderReportData>(null);
+  const [orderSummary, setOrderSummary] = useState<SummaryTableData[]>([]);
+  const [channelWiseOrderData, setChannelWiseOrderData] = useState<ChartData>(null);
+  const [paymentWiseOrderData, setPaymentWiseOrderData] = useState<ChartData>(null);
+  const [hourlyOrderTrendData, setHourlyOrderTrendData] = useState<ChartData>(null);
+  const [localeWiseHourlyOrderData, setLocaleWiseHourlyOrderData] = useState<OrderTableData[]>(null);
+  const [channelWiseHourlyOrderData, setChannelWiseHourlyOrderData] = useState<OrderTableData[]>(null);
+  const [paymentWiseHourlyOrderData, setPaymentWiseHourlyOrderData] = useState<OrderTableData[]>(null);
+  const [localeWiseTotalData, setLocaleWiseTotalData] = useState<number[]>(null);
+  const [channelWiseTotalData, setChannelWiseTotalData] = useState<number[]>(null);
+  const [paymentWiseTotalData, setPaymentWiseTotalData] = useState<number[]>(null);
+  const [tabValue, setTabValue] = useState<number>(0);
+  const { width } = useScreenSize();
+
+  const getHourlyReport = async () => {
+    try {
+      const data = await fetchData(URL_HOURLYREPORT, {});
+      setHourlyReport(data);
+    } catch (error) {
+      console.error("Error while fetching data:", error);
+    } finally {
+      setIsLoading(false);
+    }
+  };
+
+  const getOrderSummary = () => {
+    let data = hourlyReport.summary;
+    let summary = [
+      {
+        [ORDER_SUMMARY_LABELS.TOTAL_ORDER_COUNT]: data.totalOrdersPlaced,
+        [ORDER_SUMMARY_LABELS.US_ORDER_COUNT]: data.usOrdersPlaced,
+        [ORDER_SUMMARY_LABELS.CA_ORDER_COUNT]: data.caOrdersPlaced
+      },
+      {
+        [ORDER_SUMMARY_LABELS.TOTAL_SALES]: [data.totalSales, ORDER_REPORT_LABELS.USD],
+        [ORDER_SUMMARY_LABELS.US_SALES]: [data.usSales, ORDER_REPORT_LABELS.USD],
+        [ORDER_SUMMARY_LABELS.CA_SALES]: [data.caSales, ORDER_REPORT_LABELS.CAD]
+      },
+      {
+        [ORDER_SUMMARY_LABELS.TOTAL_SDD_ORDERS_PLACED]: data.sddOrdersPlaced,
+        [ORDER_SUMMARY_LABELS.TOTAL_SDD_ORDERS_COMPLETED]: data.sddOrdersCompleted,
+        [ORDER_SUMMARY_LABELS.TOTAL_BOPIS_ORDERS_PLACED]: data.bopisOrdersPlaced
+      },
+      {
+        [ORDER_SUMMARY_LABELS.AVERAGE_ORDER_VALUE]: [data.averageOrderValue, ORDER_REPORT_LABELS.USD],
+        [ORDER_SUMMARY_LABELS.US_AVERAGE_ORDER_VALUE]: [data.usAverageOrderValue, ORDER_REPORT_LABELS.USD],
+        [ORDER_SUMMARY_LABELS.CA_AVERAGE_ORDER_VALUE]: [data.caAverageOrderValue, ORDER_REPORT_LABELS.CAD]
+      },
+    ]
+    setOrderSummary(summary);
+  }
+
+  const getChannelWiseOrder = () => {
+    let data = hourlyReport.channelWiseOrderReport;
+    let datasets = [
```

Comment:
always const over let.
Check in all places.

Reply:
Added const instead of let where the variables are not reassigned